src/components/performance.cjsx

```diff
@@ -14,6 +14,29 @@ Performance = React.createClass
   propTypes:
     courseId: React.PropTypes.string.isRequired
 
+  getInitialState: ->
+    sortOrder: 'is-ascending'
+    sortIndex: 0
+    isNameSort: true
+
+  sortClick: (event) ->
+    isActiveSort = event.target.classList.contains('is-ascending', 'is-descending')
```

Comment:
its just checking classNames before the state change; it does not deform/append/remove any dom elements so it should not cause any invariants